.github/workflows/pr_checks.yml

```diff
@@ -0,0 +1,156 @@
+name: PR Checks
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths: # Only run tests when certain paths change
+      - fidesctl/**
+      - .github/workflows/test.yaml
+
+env:
+  CONTAINER: fidesctl-${{ github.event.pull_request.head.sha }}
+  IMAGE: ethyca/fidesctl:${{ github.event.pull_request.head.sha }}
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build fidesctl container
+        uses: docker/build-push-action@v2
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          context: ./fidesctl
+          file: ./fidesctl/Dockerfile
+          outputs: type=docker,dest=/tmp/${{ env.CONTAINER }}.tar
+          push: false
+          tags: ${{ env.IMAGE }}
+
+      - name: Upload fidesctl container
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.CONTAINER }}
+          path: /tmp/${{ env.CONTAINER }}.tar
+          retention-days: 7
+
+  Evaluate:
+    needs: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download fidesctl container
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ env.CONTAINER }}
+          path: /tmp/
+
+      - name: Load fidesctl image
+        run: docker load --input /tmp/${{ env.CONTAINER }}.tar
+
+      - name: Check fidesctl installation
+        run: docker run ${{ env.IMAGE }} fidesctl
+
+      - name: Run fidesctl evaluation
+        run: docker run --rm ${{ env.IMAGE }} fidesctl --local evaluate fides_resources/
+
+  Format:
+    needs: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download fidesctl container
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ env.CONTAINER }}
+          path: /tmp/
+
+      - name: Load fidesctl image
+        run: docker load --input /tmp/${{ env.CONTAINER }}.tar
+
+      - name: Run formatter
+        run: docker run ${{ env.IMAGE }} black --check src/
```

Comment:
Ideally these static analysis tools would be run in CI without having to download the container, load it into memory, and start it up. In order to make that work, the job would have to checkout the repository, install Python/`pip`, the required dependency, and execute the tool as if from a command line. This could be further optimized by [caching the dependency installations](https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows), but given the amount of time I already spent on these changes, it seemed like a decent project for another issue.